### ISSUE TITLE
Line number in sqlfluff output always seems to be 1

### ISSUE THREAD

**Issue by efung** [anboralabs/sqlfluff-intellij-community#12]
I'm using the Community plugin version 1.2.1, with sqlfluff version 1.4.5 on macOS 13.2.

When I browse the sqlfluff linting output, the line that an error occurs on always seems to be 1.

Example SQL file:

```sql
-- SELECT col
-- FOM my1table

WITH blah AS (
    SELECT col1, col2
    FROM my2table
)

SELECT
  col1,
  AVG(col2)
GROUP BY col1
ORDER BY col1
```

Here is what the CLI sqlfluff returns:

```
== [blah.sql] FAIL
L:   4 | P:   6 | L045 | Query defines CTE "blah" but does not use it.
L:   5 | P:   5 | L036 | Select targets should be on a new line unless there is
                       | only one select target.
L:  10 | P:   3 | L003 | Expected 1 indentation, found less than 1 [compared to
                       | line 09]
L:  11 | P:   3 | L003 | Expected 1 indentation, found less than 1 [compared to
                       | line 09]
L:  11 | P:   3 | L013 | Column expression without alias. Use explicit `AS`
                       | clause.
L:  12 | P:   1 |  PRS | Line 12, Position 1: Found unparsable section: 'GROUP BY
                       | col1'
```

But the sqlfluff plugin output always refers to Line 1:

<img width="606" alt="Screenshot 2023-02-01 at 12 19 47" src="https://user-images.githubusercontent.com/301094/216116565-bae99be0-1c4d-499e-bc68-5afaadfcda8a.png">

**Comment by dalgarins** [comment]
@efung which Intellij are you using? community o ultimate, this issue is related to that version and it's because community don't support sql at all and don't have any parser in the intellij API (only for intellij community).

**Comment by dalgarins** [comment]
@efung sorry, I was looking for a solution but in this moment it is not possible.

**Comment by efung** [comment]
> @efung which Intellij are you using? community o ultimate, this issue is related to that version and it's because community don't support sql at all and don't have any parser in the intellij API (only for intellij community).

I'm using IntelliJ IDEA 2022.3.1 (Ultimate Edition), Build #IU-223.8214.52, built on December 20, 2022

> I was looking for a solution but in this moment it is not possible. Intellij API has some limitations.

But the sqlfluff linting output gives you the line number where the linting violation or parse error happened. Doesn't the IntelliJ API provide a way to specify the line number that you want to report a warning on?

**Comment by dalgarins** [comment]
I can do a workaround for the IDE version you are using, and about the linter that its correct, but only show you the initial position the problem is about the length of the error, let me show you an example:

SELE * FROM users

the problem is in the SELECT keyword, sqlfluff its gonna give me the line and the position, but how can I know the length of the error only to highlight SELE as an error? in this case ok take the position that sqlfluff show and 4 spaces more to highlight that part as error, well that behavior in community version is not possible to achieve by the API limitations, instead of ultimate, in ultimate it is possible to fix.

**Comment by dalgarins** [comment]
@efung in my example I want to highlight the whole word, also I can only highlight the first letter on each issue found and with that approach could work on community versions, what do you think about that?

**Comment by efung** [comment]
For me personally, I just want the correct line number to be shown so when I double-click on the lint warning, I get taken to the right line. It matters less to me if the cursor is positioned on the exact column where the problem might be.

**Comment by dalgarins** [comment]
the issue will be fixed on version: 1.3.0

<img width="1253" alt="Captura de Pantalla 2023-02-08 a la(s) 12 06 16 a m" src="https://user-images.githubusercontent.com/8527174/217439087-32d41610-d502-4a94-97c8-f45de302e71d.png">
